### PR TITLE
Add format target and format cpp files

### PR DESCRIPTION
**Pull request by tmichalak**
Use ClangFormat to format the cpp files and add a format target to the Makefile.

### CODE REVIEW

**Comment by mithro** [comment]
@tgorochowik I like this patch! However, what format style is this using?

**Comment by tmichalak** [comment]
@mithro I forgot to add it. It is the same as in X-Ray.

**Comment by mithro** [comment]
@tmichalak -- Should probably use the Yosys config.

**Comment by mithro** [comment]
https://github.com/YosysHQ/yosys/blob/master/.clang-format

File: xdc-plugin/xdc.cc [hunk 1]
```diff
@@ -81,16 +73,18 @@ struct GetPorts : public Pass {
 		log("\n");
 		log("Get matching ports\n");
 		log("\n");
-		log("Print the output to stdout too. This is useful when all Yosys is executed\n");
+		log("Print the output to stdout too. This is useful when all "
```
Comment (mithro) [inline xdc-plugin/xdc.cc:76]: What the heck is going on here?

**Comment by litghost** [comment]
This needs a rebase?

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Default Linux style
+BasedOnStyle: LLVM
+IndentWidth: 8
+UseTab: Always
```
Comment (litghost) [inline .clang-format:4]: Please use:
```
IndentWidth: 4
TabWidth: 4
UseTab: Never
```

Comment (tmichalak) [reply .clang-format]: Fixed

**Pull request merged**